### DISCUSSION TITLE
Dragging Multiple Elements at on from the Stencil to the Paper

### DISCUSSION THREAD

**Discussion by Fractal393** [clientIO/joint#2535] [Q&A] [answered]
### Introduction

I have numerous finite elements within my stencil and I'm seeking a method to collectively drag them onto the paper, perhaps by selecting them all through a streamlined process. Is there a feasible way to accomplish this task?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment, marked as answer]
You can render a checkbox next to each stencil element (https://codepen.io/jointjs/pen/abQyLew) and when the user starts dragging any of the elements.
Then you could show some generic icon for dragging (using `dragStartClone` option).
And on `element:dragend` event, stop the current drag action (using `cancelDrag()`), and add clones of all selected elements to the main graph.

Or you can render some custom HTML stencil and use just the `ui.Stencil` API (https://codepen.io/jointjs/pen/vYVvjpW).

**Reply by Fractal393** [reply]
Thank You!
